src/tracer.js

```diff
@@ -131,12 +136,34 @@ const performPromisifyType = (err, data, type, callback) => {
   }
 };
 
+export const performStepFunctionLogic = handlerReturnValue => {
+  return (
+    safeExecute(() => {
+      if (!isStepFunction()) {
+        return handlerReturnValue;
+      }
```

Comment:
If it's not step function don't call this method, it's safer and cleaner then a function that does nothing in some cases